### PR TITLE
Adding npm scripts

### PR DESCRIPTION
**Pull request by rorticus**
npm scripts to replace grunt-dojo2! See README for full details, and https://github.com/dojo/widget-core/compare/master...rorticus:gruntless for implementation details!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,177 @@
...
+
+A safe release is a clean release. To check if there are no uncommitted changes, and the user is `master`, run the`dojo-repo-is-clean-check` script. The script will fail with a `1` exit code if the repo is dirty.
+
+#### Release
+
+The `dojo-release` script can release a dojo package. The `dist/release` directory is what gets released. The script takes a number of arguments:
+
+| Parameter  | Description                              |
+| ---------- | ---------------------------------------- |
+| `—release` | The version to release                   |
```
Comment (agubler) [inline README.md]: Perhaps we need to add an `--initial` flag that adds `--access public` to the publish command or maybe we can just include that all the time (for every publish).

File: README.md [hunk 2]
```diff
@@ -1 +1,177 @@
...
+| `—next`    | The next version (`package.json` version gets set to this) |
+| `—dry-run` | Shows the commands that will be run but does not run the commands. |
+| `—tag`     | The tag to pass to `npm publish`         |
```
Comment (agubler) [inline README.md]: Can we document the default `tag`?

File: README.md (hunk 1 above)
Comment (agubler) [reply README.md]: Also for `--initial` we'll need to skip the check against the existing maintainers (because there won't be any)

**Comment by agubler** [comment]
Fancy adding prettier to this repository?

File: src/release.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+		.filter((remote: string) => remote.trim() === 'dojo').length === 1;
+
+	console.log(chalk.yellow(`Version: ${releaseVersion}`));
+	console.log(chalk.yellow(`Next Version: ${nextVersion}`));
+	console.log(chalk.yellow(`Dry Run: ${dryRun}`));
+	console.log(chalk.yellow(`Push Back: ${hasDojoRemote}`));
+
+	// update the version
+	await command('npm', ['version', releaseVersion, '--no-git-tag-version'], { cwd: 'dist/release' }, false);
+	await command('npm', ['version', releaseVersion, '--no-git-tag-version'], {}, false);
```
Comment (agubler) [inline src/release.ts]: If we remove `--no-git-tag-version` (apart from dry runs) then it should do the git tag for us (and then we won't need the `git tag` command below.

File: README.md (hunk 1 above)
Comment (rorticus) [reply README.md]: Added an `--initial` (default false) parameter to the release script. The can-publish checks are actually done by the `dojo-can-publish-check` script. For the initial release people can just not run this script?

File: README.md (hunk 2 above)
Comment (rorticus) [reply README.md]: Documented

File: src/release.ts (hunk 3 above)
Comment (rorticus) [reply src/release.ts]: Yeah, so I did this on purpose. I didn't want the tag to be created if the `npm publish` task didn't work.

**Pull request merged**